### PR TITLE
Add EmptyState, and form button override

### PR DESCRIPTION
**Pull request by ddonahue007**
Add the ability to not render form buttons
Add ErrorState that allows an easy way to render error page.

Example:
Screen Shot 2022-01-18 at 5 56 29 PM

### CODE REVIEW

**Review by Hyperkid123 — CHANGES_REQUESTED** [review]
@ddonahue007 there is an error state component in the FEC library. We should use that instead to be consistent across services.
https://github.com/RedHatInsights/frontend-components/tree/master/packages/components/src/ErrorState

Here is an example usage: https://github.com/RedHatInsights/settings-frontend/blob/master/src/SmartComponents/Applications/Applications.js#L5

Can you also fix the snapshots?

**Comment by ddonahue007** [comment]
Here is the json data being passed to setting-frontend(pretty print)
```
[
    {
        "fields": [
            {
                "name": "middleware.settings.err",
                "component": "error-state",
                "errorTitle": "Configuration Error",
                "errorDescription": "Before adding settings you must create a Source for Cost Management.",
                "hrefUrl": "https://access.redhat.com/documentation/en-us/cost_management_service/2021//",
                "hrefText": "[Learn more]"
            }
        ],
        "formProps": {
            "showFormControls": false
        }
    }
]
```

**Review by aneelac22 — APPROVED** [review]
LGTM

File: src/PresentationalComponents/ErrState.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      }
+    />
+  );
+};
+
+ErrState.propTypes = {
+  errorTitle: propTypes.string,
+  errorDescription: propTypes.string,
+  hrefUrl: propTypes.string,
+  hrefText: propTypes.string,
```
Comment (Hyperkid123) [inline src/PresentationalComponents/ErrState.js:27]: :thinking: Maybe we could lift the description composition to the schema and use the `dangerouslySetInnerHTML` React prop. that can open a can of worms but I would assume if we manage the API response then we should be fine on that front.

Then you would use the ErrorState like so:
```jsx

export const ErrState = ({
  errorTitle,
  errorDescription,
}) => {
  return (
    <ErrorState
      errorTitle={errorTitle}
      errorDescription={
        <div dangerouslySetInnerHTML={{__html: errorDescription}}></div>
      }
    />
  );
};

<ErrState errorTitle="Foo" errorDescription="<span>Foo bar <a href="#">baz</a></span>"  />
```

**Review by Hyperkid123 — COMMENTED** [review]
@ddonahue007 got one more topic to dicuss

File: src/PresentationalComponents/ErrState.js (hunk 1 above)
Comment (ddonahue007) [reply src/PresentationalComponents/ErrState.js]: Nice suggestion, I pushed up the changes

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**